lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -127,6 +127,25 @@ defmodule ExUnit.AssertionsTest do
     end
   end
 
+  test "assert received with multiple pinned variables" do
+    status = :valid
+    send self(), {:status, :invalid, :invalid}
+    try do
+      "This should never be tested" = assert_received {
+        :status,
+        ^status,
+        ^status
```

Comment:
We need the test to show two different variables otherwise this test is not really checking the bug you have originally fixed. :)